src/overlays/actors/ovl_En_Ssh/z_en_ssh.c

```diff
@@ -37,139 +37,900 @@ const ActorInit En_Ssh_InitVars = {
+        blureInit.p2StartColor[i] = sP2StartColor[i];
+        blureInit.p1EndColor[i] = sP1EndColor[i];
+        blureInit.p2EndColor[i] = sP2EndColor[i];
+    }
+    blureInit.elemDuration = 6;
+    blureInit.unkFlag = false;
+    blureInit.calcMode = 3;
+
+    Effect_Add(globalCtx, &blureIdx, 1, 0, 0, &blureInit);
+    return blureIdx;
+}
+
+s32 EnSsh_CheckCeilingPos(EnSsh* this, GlobalContext* globalCtx) {
+    CollisionPoly* poly;
+    s32 bgId;
+    Vec3f posB;
+
+    posB.x = this->actor.world.pos.x;
+    posB.y = this->actor.world.pos.y + 1000.0f;
+    posB.z = this->actor.world.pos.z;
+    if (!BgCheck_EntityLineTest1(&globalCtx->colCtx, &this->actor.world.pos, &posB, &this->ceilingPos, &poly, 0, 0, 1,
+                                 1, &bgId)) {
+        return false;
+    }
+    return true;
+}
+
+void EnSsh_AddBlureVertex(EnSsh* this) {
+    Vec3f p1Base = { 834.0f, 834.0f, 0.0f };
+    Vec3f p2Base = { 834.0f, -584.0f, 0.0f };
+    Vec3f p1;
+    Vec3f p2;
+
+    p1Base.x *= this->colliderScale;
+    p1Base.y *= this->colliderScale;
+    p1Base.z *= this->colliderScale;
+
+    p2Base.x *= this->colliderScale;
+    p2Base.y *= this->colliderScale;
+    p2Base.z *= this->colliderScale;
+
+    Matrix_StatePush();
+    Matrix_MultiplyVector3fByState(&p1Base, &p1);
+    Matrix_MultiplyVector3fByState(&p2Base, &p2);
+    Matrix_StatePop();
+    EffectBlure_AddVertex(Effect_GetByIndex(this->blureIdx), &p1, &p2);
+}
+
+void EnSsh_AddBlureSpace(EnSsh* this) {
+    EffectBlure_AddSpace(Effect_GetByIndex(this->blureIdx));
+}
+
+void EnSsh_InitColliders(EnSsh* this, GlobalContext* globalCtx) {
+    ColliderCylinderInit* cylinders[] = {
+        &sCylinderInit1, &sCylinderInit1, &sCylinderInit1, &sCylinderInit2, &sCylinderInit2, &sCylinderInit2,
+    };
+    s32 i;
+    s32 pad;
+
+    for (i = 0; i < ARRAY_COUNT(this->collider1); i++) {
+        Collider_InitAndSetCylinder(globalCtx, &this->collider1[i], &this->actor, cylinders[i]);
+    }
+
+    this->collider1[0].info.bumper.dmgFlags = 0x38A9;
+    this->collider1[1].info.bumper.dmgFlags = ~0x83038A9;
+    this->collider1[2].base.colType = COLTYPE_METAL;
+    this->collider1[2].info.bumperFlags = (BUMP_NO_AT_INFO | BUMP_HOOKABLE | BUMP_ON);
+    this->collider1[2].info.elemType = ELEMTYPE_UNK2;
+    this->collider1[2].info.bumper.dmgFlags = ~0x83038A9;
+
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(2), &sColChkInfoInit);
+    Collider_InitJntSph(globalCtx, &this->collider2);
+    Collider_SetJntSph(globalCtx, &this->collider2, &this->actor, &sJntSphInit, this->collider2Elements);
+}
+
+f32 EnSsh_SetAnimation(EnSsh* this, s32 arg0) {
+    AnimationHeader* animation[] = { &object_ssh_Anim_006D78, &object_ssh_Anim_001494, &object_ssh_Anim_001494,
+                                     &object_ssh_Anim_006788, &object_ssh_Anim_001494, &object_ssh_Anim_001494,
+                                     &object_ssh_Anim_006D78 };
```

Comment:
```suggestion
    AnimationHeader* animation[] = { &object_ssh_Anim_006D78, &object_ssh_Anim_001494, &object_ssh_Anim_001494,
                                     &object_ssh_Anim_006788, &object_ssh_Anim_001494, &object_ssh_Anim_001494,
                                     &object_ssh_Anim_006D78, };
```
Can you add the trailing comma and reformat?